packages/redis/src/RedisClient.ts

```diff
@@ -214,4 +250,289 @@ export class RedisClient implements IRedisClient, IRequireInitialization, IDispo
             span.finish();
         }
     }
+
+    public async xReadObject<T>(
+        context: SpanContext,
+        type: string | IClassType<T>,
+        streamName: string,
+        id?: string
+    ): Promise<[string, T] | undefined> {
+        const db = this.config.db;
+        const span = this.tracer.startSpan(this.spanOperationName, { childOf: context });
+        this.spanLogAndSetTags(span, this.xReadObject.name, this.config.db, id, streamName);
+        try {
+            const typeName = this.getTypeName(type);
+            const response = await this.client.xread(["block", "0", "streams", streamName, id]);
+            const value = extractStreamValue(response);
+            const streamId = extractStreamId(response);
+            let data;
+            if (value) {
+                const buf = this.config.base64Encode
+                    ? Buffer.from(value, "base64")
+                    : Buffer.from(value);
+                const msg = this.encoder.decode(new Uint8Array(buf), typeName);
+                data = msg.payload;
+            }
+
+            this.metrics.increment(RedisMetrics.XRead, {
+                type,
+                db,
+                result: RedisMetricResults.Success,
+            });
+            return [streamId, data];
+        } catch (e) {
+            failSpan(span, e);
+            this.metrics.increment(RedisMetrics.XRead, {
+                db,
+                result: RedisMetricResults.Error,
+                error: e,
+            });
+            throw e;
+        } finally {
+            span.finish();
+        }
+    }
+
+    public async xGroup(
+        context: SpanContext,
+        streamName: string,
+        consumerGroup: string,
+        supressAlreadyExistsError: boolean = true
+    ): Promise<string> {
+        const db = this.config.db;
+        const span = this.tracer.startSpan(this.spanOperationName, { childOf: context });
```

Comment:
It looks like all of the operations in this class use the same `spanOperationName`. IMO the resulting trace data would be much easier to understand if instead the child spans were named after their respective Redis commands.